src/test/java/uk.gov.ons.census.action/messaging/ActionFulfilmentIT.java

```diff
@@ -0,0 +1,126 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.jeasy.random.EasyRandom;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.census.action.model.dto.*;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.repository.CaseRepository;
+
+@ContextConfiguration
+@SpringBootTest
+@ActiveProfiles("test")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@RunWith(SpringJUnit4ClassRunner.class)
+public class ActionFulfilmentIT {
+
+  @Value("${queueconfig.action-fulfilment-inbound-queue}")
+  private String actionFulfilmentQueue;
+
+  @Value("${queueconfig.events-exchange}")
+  private String eventsExchange;
+
+  @Value("${queueconfig.events-fulfilment-request-binding}")
+  private String eventsFulfilmentRequestBinding;
+
+  @Value("${queueconfig.outbound-printer-queue}")
+  private String outboundPrinterQueue;
+
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(wireMockConfig().port(8089).httpsPort(8443));
+
+  @Autowired private RabbitQueueHelper rabbitQueueHelper;
+
+  @Autowired private CaseRepository caseRepository;
+
+  private EasyRandom easyRandom = new EasyRandom();
+
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  @Before
+  @Transactional
+  public void setUp() {
+    rabbitQueueHelper.purgeQueue(actionFulfilmentQueue);
+    rabbitQueueHelper.purgeQueue(outboundPrinterQueue);
+  }
+
+  @Test
+  public void checkReceivedEventsAreEmitted() throws InterruptedException, IOException {
+    BlockingQueue<String> outputQueue = rabbitQueueHelper.listen(outboundPrinterQueue);
+    Case fulfillmentCase = this.setUpCase();
+    // Given
+    ResponseManagementEvent actionFulfilmentEvent =
+        getResponseManagementEvent(fulfillmentCase.getCaseId().toString());
+    PrintFileDto printFileRequest = setupPrintFile();
+    String url = "/uacqid/create/";
+    UacQidDTO uacQidDto = easyRandom.nextObject(UacQidDTO.class);
+    String returnJson = objectMapper.writeValueAsString(uacQidDto);
+    givenThat(
```

Comment:
Small question(!) - I understand this will just give us a UAC/QID link, but is it worth actually calling this endpoint since we're using it in an IT?